src/algo/tensor/tensor.py

```diff
@@ -0,0 +1,84 @@
+from pygraphblas import *
+from abc import ABC
+
+from src.graph.label_graph import LabelGraph
+from src.grammar.rsa import RecursiveAutomaton
+
+
+class TensorSolver(ABC):
```

Comment:
The main purpose of introducing an abstract class is to create an abstract _solver_ function. This will allow you to implement different versions of the algorithms, which based on the same type of input data and return the same type of result. This leads to the possibility of processing different algorithms in the same way.
In your case, this function has no arguments, so I think either TensorSolver should not exist (or at least it shouldn't be abstract), or it should have an abstract _solve_ function.
I think the best solution is making solve the abstract method of TensorSolver with the following signature:
`solve(graph :LabelGraph, grammar: RecursiveAuotomaton) -> SomeReturnType`